app.py

```diff
@@ -115,119 +120,23 @@ def get_frame_range_from_shotgun(self):
 
         return ( data[sg_in_field], data[sg_out_field] )
 
-
     def get_current_frame_range(self, engine):
+        result = self.execute_hook_method("hook_frame_operation", "get_frame_range")
 
-        if engine == "tk-maya":
-            import pymel.core as pm
-            import maya.cmds as cmds
-            current_in = cmds.playbackOptions(query=True, minTime=True)
-            current_out = cmds.playbackOptions(query=True, maxTime=True)
-
-        elif engine == "tk-nuke" and not self.engine.hiero_enabled:
-            import nuke
-            current_in = int(nuke.root()["first_frame"].value())
-            current_out = int(nuke.root()["last_frame"].value())
-
-        elif engine == "tk-motionbuilder":
-            from pyfbsdk import FBPlayerControl, FBTime
-
-            lPlayer = FBPlayerControl()
-            current_in = lPlayer.LoopStart.GetFrame()
-            current_out = lPlayer.LoopStop.GetFrame()
-
-        elif engine == "tk-softimage":
-            import win32com
-            xsi = win32com.client.Dispatch('XSI.Application')
-
-            current_in = xsi.GetValue("PlayControl.In")
-            current_out = xsi.GetValue("PlayControl.Out")
-
-        elif engine == "tk-houdini":
-            import hou
-            current_in, current_out = hou.playbar.playbackRange()
-
-        elif engine == "tk-3dsmax":
-            from Py3dsMax import mxs
-            current_in = mxs.animationRange.start
-            current_out = mxs.animationRange.end
-        elif engine == "tk-3dsmaxplus":
-            import MaxPlus
-            ticks = MaxPlus.Core.EvalMAXScript("ticksperframe").GetInt()
-            current_in = MaxPlus.Animation.GetAnimRange().Start() / ticks
-            current_out = MaxPlus.Animation.GetAnimRange().End() / ticks
-
-        else:
-            raise tank.TankError("Don't know how to get current frame range for engine %s!" % engine)
-
-        return (current_in, current_out)
+        if not isinstance(result, tuple) or (isinstance(result, tuple) and len(result) != 2):
+            raise tank.TankError(
+                "Unexpected type returned from 'hook_frame_operation' for operation get_"
+                "frame_range - expected a 'tuple' with (in_frame, out_frame) values but "
+                "returned '%s' : %s" % (type(result).__name__),
+                result
+            )
+        return result
 
     def set_frame_range(self, engine, in_frame, out_frame):
```

Comment:
And a docstring here too.

```suggestion
     def set_frame_range(self, engine, in_frame, out_frame):
        """
        Sets the current scene frame range.

        :param engine: The current engine.
        :param int in_frame: in_frame for the current context
            (e.g. the current shot, current asset etc)
        :param int out_frame: out_frame for the current context
            (e.g. the current shot, current asset etc)

        :raises TankError: Raised if an error is encountered when setting the frame range for the current engine.
        """
```

Reply:
Done